server/tests/riders.test.ts

```diff
@@ -0,0 +1,261 @@
+    description: '',
+    joinDate: '2023-03-09',
+    endDate: '2024-03-09',
+    address: '37 Colonial Ln, Ithaca, NY 14850',
+    favoriteLocations: ['west'],
+    organization: 'CULift',
+    photoLink: '',
+    active: true,
+  },
+];
+
+describe('Riders', () => {
+  before(async () => {
+    adminToken = await authorize('Admin', {
+      firstName: 'Test-Admin',
+      lastName: 'Test-Admin',
+      phoneNumber: '1111111111',
+      email: 'test-admin@cornell.edu',
+    });
+    await populateDB(Rider, {
+      id: 'abc-10',
+      email: 'test-email@test.com',
+      phoneNumber: '1234567890',
+      firstName: 'Test',
+      lastName: 'Testing',
+      pronouns: 'he/him/his',
+      accessibility: 'Crutches',
+      description: '',
+      joinDate: '2023-03-09',
+      endDate: '2024-03-09',
+      address: '36 Colonial Ln, Ithaca, NY 14850',
+      favoriteLocations: ['west'],
+      organization: 'CULift',
+      photoLink: '',
+      active: true,
+    });
+    // add second rider
+    await populateDB(Rider, {
+      id: 'abc-11',
+      email: 'test-email1@test.com',
+      phoneNumber: '1234567891',
+      firstName: 'Test',
+      lastName: 'Testing1',
+      pronouns: 'he/him/his',
+      accessibility: 'Crutches',
+      description: '',
+      joinDate: '2023-03-09',
+      endDate: '2024-03-09',
+      address: '37 Colonial Ln, Ithaca, NY 14850',
+      favoriteLocations: ['west'],
+      organization: 'CULift',
+      photoLink: '',
+      active: true,
+    });
+  });
+
+  after(clearDB);
+
+  // testing retrieval of specific rider
+  describe('GET /api/riders/abc-10', () => {
+    it('should return 200 OK', async () => {
+      const res = await request(app)
+        .get('/api/riders/abc-10')
+        .set('Authorization', `Bearer ${adminToken}`)
+        .expect(200)
+        .expect('Content-Type', 'application/json; charset=utf-8');
+      expect(res.status).to.be.equal(200);
+      expect(res.body).to.have.property('data');
+      expect(res.body.data).to.deep.equal(riders[0]);
+    });
+  });
+
+  // testing retrieval of all riders
+  describe('GET /api/riders', () => {
+    it('should return 200 OK', async () => {
+      const res = await request(app)
+        .get('/api/riders/')
+        .set('Authorization', `Bearer ${adminToken}`)
+        .expect(200)
+        .expect('Content-Type', 'application/json; charset=utf-8');
+      expect(res.status).to.be.equal(200);
+      expect(res.body).to.have.property('data');
+      expect(res.body.data.reverse()).to.deep.equal(riders);
+    });
+  });
+
+  // testing retrieval of rider profile
+  describe('GET /api/riders/abc-10/profile', () => {
+    it('should return 200 OK', async () => {
+      const res = await request(app)
+        .get('/api/riders/abc-10/profile')
+        .set('Authorization', `Bearer ${adminToken}`)
+        .expect(200)
+        .expect('Content-Type', 'application/json; charset=utf-8');
+      expect(res.status).to.be.equal(200);
```

Comment:
would recommend using `deep`
1. remove the fields from the rider that are not returned and do deep equals expRider